py3status/modules/window_title.py

```diff
@@ -42,20 +45,36 @@ class Py3status:
     format = '{title}'
     max_width = 120
 
+    def post_config_hook(self):
+        # empty defaults to replace window properties
+        self.empty_defaults = {
+            x: '' for x in self.py3.get_placeholders_list(self.format)}
+        self.empty_defaults.update({'title': ''})
+
+    def _find_focused(self, tree):
+        if isinstance(tree, list):
+            for el in tree:
+                res = self._find_focused(el)
+                if res:
+                    return res
+        elif isinstance(tree, dict):
+            if tree['focused']:
+                return tree
+            return self._find_focused(tree['nodes'] + tree['floating_nodes'])
+        return {}
+
     def window_title(self):
         tree = loads(self.py3.command_output('i3-msg -t get_tree'))
-        window = find_focused(tree)
+        window_properties = self._find_focused(tree).get(
+            'window_properties', self.empty_defaults)
 
-        if not window or window.get('name') is None or window.get('type') == 'workspace':
-            title = ''
-        elif len(window['name']) > self.max_width:
-            title = u"...{}".format(window['name'][-(self.max_width - 3):])
-        else:
-            title = window['name']
+        if len(window_properties['title']) > self.max_width:
```

Comment:
`len(window_properties.get('title', ''))` is nicer

Reply:
Ya. Let's merge `hueshift` next.